app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java

```diff
@@ -119,4 +130,18 @@ public void getShuttle(View view) {
         String content = mViewModel.getShuttleDisplayText(shuttles);
         this.content.setText(content);
     }
+
+    public void getRouteOptions() {
+        Coordinates fromCenterCoordinates = mViewModel.getFrom().getCenterCoordinates();
+        Coordinates toCenterCoordinates = mViewModel.getTo().getCenterCoordinates();
+
+        if(fromCenterCoordinates != null && toCenterCoordinates != null) {
+            LatLng from = new LatLng(fromCenterCoordinates.getLatitude(), fromCenterCoordinates.getLongitude());
+            LatLng to = new LatLng(toCenterCoordinates.getLatitude(), toCenterCoordinates.getLongitude());
+            String transitType = ClassConstants.TRANSIT; // TODO: #163: dynamically change the transit type depending on the mode of transportation icon clicked on the RoutesActivity page. By default, it will be TRANSIT.
+
+            String url = UrlBuilder.build(from, to, transitType);
```

Comment:
this code inside the if statement could go inside the viewmodel and be testable no?

Reply:
you are right!

Reply:
I already created a unit test for the build method to ensure it builds the correct url!

Reply:
It's on my unit tests branch tho